underfs/oss/src/main/java/alluxio/underfs/oss/OSSUnderFileSystem.java

```diff
@@ -458,59 +148,105 @@ private boolean copy(String src, String dst) {
     }
   }
 
-  /**
-   * Internal function to delete a key in OSS.
-   *
-   * @param key the key to delete
-   * @return true if successful, false if an exception is thrown
-   */
-  private boolean deleteInternal(String key) {
+  @Override
+  protected OutputStream createObject(String key) throws IOException {
+    return new OSSOutputStream(mBucketName, key, mClient);
+  }
+
+  @Override
+  protected boolean deleteObject(String key) {
     try {
-      if (isDirectory(key)) {
-        String keyAsFolder = convertToFolderName(stripPrefixIfPresent(key));
-        mClient.deleteObject(mBucketName, keyAsFolder);
-      } else {
-        mClient.deleteObject(mBucketName, stripPrefixIfPresent(key));
-      }
+      mClient.deleteObject(mBucketName, key);
     } catch (ServiceException e) {
       LOG.error("Failed to delete {}", key, e);
       return false;
     }
     return true;
   }
 
+  @Override
+  protected String getFolderSuffix() {
+    return FOLDER_SUFFIX;
+  }
+
+  @Override
+  protected ObjectListingResult getObjectListing(String path, boolean recursive)
+      throws IOException {
+    return new GCSObjectListingResult(path, recursive);
+  }
+
   /**
-   * @param key the key to get the object details of
-   * @return {@link ObjectMetadata} of the key, or null if the key does not exist
+   * Wrapper over GCS {@link StorageObjectsChunk}.
    */
-  private ObjectMetadata getObjectDetails(String key) {
+  final class GCSObjectListingResult implements ObjectListingResult {
```

Comment:
Should this be `OSSObjectListingResult`? Similar comments to `GCSObjectListingResult`